environ/environ.py

```diff
@@ -64,6 +65,8 @@ def _cast_int(v):
 def _cast_urlstr(v):
     return unquote(v) if isinstance(v, str) else v
 
+def _urlparse_quote(url):
+    return urlparse(quote(url, safe=':/?&=@'))
```

Comment:
flake8 expected 2 blank lines here, found 1

```suggestion
    return urlparse(quote(url, safe=':/?&=@'))

```